demos/demo00_MinimalExample.py

```diff
@@ -4,49 +4,46 @@
-model      = LinearModel(A)                           #Model for inverse problem
-prior      = Gaussian(np.zeros(n),0.1)                #Prior distribution
-likelihood = Gaussian(model,0.05).to_likelihood(b)    #Likelihood function
-IP         = BayesianProblem(likelihood, prior)        #Bayesian model for inverse problem
-IP.UQ(exact=x_exact)                                  #Perform UQ on inverse problem
+A  = LinearModel(Amat)                          # y = Ax. Model for inverse problem
+x  = Gaussian(np.zeros(n), 0.1)                 # x ~ N(0,0.1)
+y  = Gaussian(A@x, 0.05)                        # y ~ N(Ax,0.05)
+IP = BayesianProblem(y, x).set_data(y=y_data)   # Bayesian problem given observed data
+IP.UQ(exact=x_exact)                            # Run UQ analysis
 
 # %%
 # Wrap into CUQI "testproblem".
-TP = cuqi.testproblem.Deconvolution1D(prior=prior)
+TP = cuqi.testproblem.Deconvolution1D(prior=x, phantom="square")
 TP.UQ()
 
 # %%
 # switch prior
 # Set up Bayesian model for inverse problem
-M  = LinearModel(forward=A)                                          #Model for inverse problem
-P  = Cauchy_diff(location=np.zeros(n),scale=0.05,bc_type='neumann')  #Prior distribution
-L  = Gaussian(mean=M,std=0.05).to_likelihood(data=b)                 #Likelihood function
-IP = BayesianProblem(likelihood=L, prior=P)                          #Bayesian model for inverse problem
-IP.UQ(exact=x_exact)                                                 #Perform UQ on inverse problem
+A  = LinearModel(Amat)                                  # y = Ax. Model for inverse problem
+x  = Laplace_diff(np.zeros(n), 0.01, bc_type='zero')    # x ~ Laplace_diff(0,0.01), Zero BC
+y  = Gaussian(A@x, 0.05)                                # y ~ N(Ax,0.05)
+IP = BayesianProblem(y, x).set_data(y=y_data)           # Bayesian problem given observed data
+IP.UQ(exact=x_exact)                                    # Run UQ analysis
 
 # %%
-samples = IP.sample_posterior(50000)
-# %%
-samples.plot_ci(95,exact=x_exact)
-# %%
+# switch prior again
 # Set up Bayesian model for inverse problem
-model      = LinearModel(A)                             #Model for inverse problem
-prior      = Cauchy_diff(np.zeros(n),0.05,'neumann')    #Prior distribution
-likelihood = Gaussian(model,0.05).to_likelihood(data=b) #Likelihood function
-IP         = BayesianProblem(likelihood,prior)          #Bayesian model for inverse problem
-IP.UQ()                                                 #Perform UQ on inverse problem        
\ No newline at end of file
+A  = LinearModel(Amat)                                  # y = Ax. Model for inverse problem
+x  = Cauchy_diff(np.zeros(n), 0.01, bc_type='zero')     # x ~ Cauchy_diff(0,0.01), Zero BC
+y  = Gaussian(A@x, 0.05)                                # y ~ N(Ax,0.05)
+IP = BayesianProblem(y, x).set_data(y=y_data)           # Bayesian problem given observed data
+IP.UQ(exact=x_exact)                                    # Run UQ analysis
```

Comment:
Same here, may be can return in samples for illustration 
```suggestion
samples = IP.UQ(exact=x_exact)                            # Run UQ analysis
```